lib/pos_blockchain.go

```diff
@@ -1128,12 +1440,24 @@ func (bc *Blockchain) shouldReorg(blockNode *BlockNode, currentView uint64) bool
 	return blockNode.Header.ProposedInView >= currentView
 }
 
-// addBlockToBestChain adds the block to the best chain.
-func (bc *Blockchain) addBlockToBestChain(blockNode *BlockNode) {
+// addTipBlockToBestChain adds the block as the new tip of the best chain.
+func (bc *Blockchain) addTipBlockToBestChain(blockNode *BlockNode) {
 	bc.bestChain = append(bc.bestChain, blockNode)
 	bc.bestChainMap[*blockNode.Hash] = blockNode
 }
 
+// removeTipBlockFromBestChain removes the current tip from the best chain. It
+// naively removes the tip regardless of the tip's status (committed or not).
+// This function is a general purpose helper function that bundles mutations to
+// the bestChain slice and bestChainMap map.
+func (bc *Blockchain) removeTipBlockFromBestChain() *BlockNode {
```

Comment:
So I think we're modifying the bestHeaderChain independent of the bestChain, even though we want them to be in-sync. I think it's fine, just feels a bit scary to have two things that need to remain consistent being modified independently.

Reply:
Hmm, I'm not sure this is much of an issue. I did notice in ProcessBlockPoW that we cann processHeaderPoW separately, which mutates the bestHeaderChain separately so I figured this would be safe.

I'll noodle on this to see if we can better couple mutations to the bestChain and bestHeaderChain here while keeping the reorgs simple. LMK if you have a strong opinion on this one

Reply:
On second review, updating the bestChain in lockstep with the bestHeaderChain here is fine. It's turns out to be a no-op from the context of running processBlockPoS because the bestHeaderChain always ends up with the same final state as the bestChain. But the peace of mind of updating them in lockstep here is a nice to have. I've addressed your feedback in https://github.com/deso-protocol/core/pull/962